src/NosCore.Controllers/ExchangePacketController.cs

```diff
@@ -0,0 +1,251 @@
+using NosCore.Configuration;
+using NosCore.GameObject;
+using NosCore.GameObject.ComponentEntities.Extensions;
+using NosCore.GameObject.ComponentEntities.Interfaces;
+using NosCore.GameObject.Networking;
+using NosCore.GameObject.Services.ExchangeInfo;
+using NosCore.GameObject.Services.ItemBuilder;
+using NosCore.GameObject.Services.ItemBuilder.Item;
+using NosCore.Packets.ClientPackets;
+using NosCore.Packets.ServerPackets;
+using NosCore.Shared.Enumerations;
+using NosCore.Shared.Enumerations.Buff;
+using NosCore.Shared.Enumerations.Character;
+using NosCore.Shared.Enumerations.Interaction;
+using NosCore.Shared.Enumerations.Items;
+using NosCore.Shared.I18N;
+using Serilog;
+
+namespace NosCore.Controllers
+{
+    public class ExchangePacketController : PacketController
+    {
+        private readonly WorldConfiguration _worldConfiguration;
+        private readonly ExchangeAccessService _exchangeAccessService;
+        private readonly ILogger _logger = Logger.GetLoggerConfiguration().CreateLogger();
+        
+        public ExchangePacketController(WorldConfiguration worldConfiguration, ExchangeAccessService exchangeAccessService)
+        {
+            _worldConfiguration = worldConfiguration;
+            _exchangeAccessService = exchangeAccessService;
+        }
+
+        [UsedImplicitly]
+        public ExchangePacketController()
+        {
+
+        }
+
+        [UsedImplicitly]
+        public void ExchangeList(ExcListPacket packet)
+        {
+            if (packet.Gold > Session.Character.Gold || packet.BankGold > Session.Account.BankMoney)
+            {
+                return;
+            }
+
+            var list = new ServerExcListPacket
+            {
+                SenderType = SenderType.Server,
+                VisualId = Session.Character.VisualId,
+                Gold = packet.Gold,
+                BankGold = packet.BankGold,
+                SubPackets = new List<ServerExcListSubPacket>()
+            };
+            
+            var target = Broadcaster.Instance.GetCharacter(s => s.VisualId == Session.Character.ExchangeData.TargetVisualId && s.MapInstanceId == Session.Character.MapInstanceId) as Character;
+
+
+            if (packet.SubPackets.Any())
+            {
+                byte i = 0;
+                foreach (var value in packet.SubPackets)
+                {
+                    if (value.PocketType == PocketType.Bazaar)
+                    {
+                        return;
+                    }
+
+                    var item = Session.Character.Inventory.LoadBySlotAndType<IItemInstance>(value.Slot, value.PocketType);
+
+                    if (item == null || item.Amount <= 0 || item.Amount < value.Amount)
+                    {
+                        return;
+                    }
+
+                    if (!item.Item.IsTradable)
+                    {
+                        _exchangeAccessService.CloseExchange(Session, target?.Session, ExchangeCloseType.Failure);
```

Comment:
doesnt seems correct to pass a potential null. it would be better closing if null before